sklearn/preprocessing/_encoders.py

```diff
@@ -367,6 +384,14 @@ def _validate_keywords(self):
                 )
             )
             raise ValueError(msg)
+        if isinstance(
+            self.feature_name_combiner, str
+        ) and self.feature_name_combiner not in ("concat_string"):
```

Comment:
We can call `_get_feature_name_combiner` here to do the error checking and not use the return.